gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/AbstractJWTFilter.java

```diff
@@ -342,7 +376,34 @@ protected boolean validateToken(HttpServletRequest request, HttpServletResponse
     return false;
   }
 
-  protected boolean verifyTokenSignature(final JWT token) {
+  protected boolean validateToken(final HttpServletRequest request,
+                                  final HttpServletResponse response,
+                                  final FilterChain chain,
+                                  final String tokenId)
+          throws IOException, ServletException {
+
+    if (tokenStateService != null) {
+      try {
+//        TokenMetadata metadata = tokenStateService.getTokenMetadata(tokenId);
+//      if (expectedIssuer.equals(metadata.getIssuer())) { // TODO: PJZ: Additional metadata support for validation
```

Comment:
remove commented out code?

Reply:
I see - future metadata...
I don't know that it makes sense to validate the issuer unless it has come in via JWT.
If it is a known token then we are obviously the issuer.
Audience will make sense though.

Reply:
Yes, I've created KNOX-2563 to provide the necessary metadata, so the comments can go.